tests/Issues/ScopeNotAvailable/ForeachValueScopeTest.php

```diff
@@ -6,13 +6,15 @@
 
 use Iterator;
 use PHPUnit\Framework\Attributes\DataProvider;
+use Rector\Exception\Configuration\RectorRuleShouldNotBeAppliedException;
 use Rector\Testing\PHPUnit\AbstractRectorTestCase;
 
 final class ForeachValueScopeTest extends AbstractRectorTestCase
 {
     #[DataProvider('provideData')]
     public function test(string $filePath): void
     {
+        $this->expectException(RectorRuleShouldNotBeAppliedException::class);
```

Comment:
In this particular test, the `refactor()` function returns on purpose a Node element even though it does not change it. This is done to confirm that the code that handles this node does not throw an error. So in this particular case we need to confirm that we know that this rule would be applied even though it does not change the code